File: Pathfinder.Web/ClientApp/src/models/Product.js
class Product {
  constructor() {
    this.id = 0;
    this.name = "";
    this.description = "";
    this.categoryType = 0;
    this.price = 0;
    this.category = null;
    this.copyModel = target => {
      this.id = target.id;
      this.categoryId = target.category?.id;
      this.description = target.description;
      this.name = target.name;
      this.price = Number.parseFloat(target.price);
    };
  }
}
export default Product;

File: Pathfinder.Core/Entities/Product.cs
﻿using Pathfinder.Core.Entities.Base;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Pathfinder.Core.Entities
{
    public class Product : Entity
    {
        [Required, StringLength(80)]
        public string Name { get; set; }
        public string Description { get; set; }
        public string ImageFile { get; set; }
        public decimal? Price { get; set; }

        // n-1 relationships
        public int CategoryId { get; set; }
        public virtual Category Category { get; set; }
    }
}
